src/hawkbit-client.c

```diff
@@ -914,6 +914,17 @@ static gpointer download_thread(gpointer data)
         }
 
         g_mutex_lock(&active_action->mutex);
+        // skip installation if hawkBit asked us to do so
+        if (!artifact->do_install &&
+            (!artifact->do_maintenance || g_strcmp0(artifact->do_maintenance, "available") == 0)) {
```

Comment:
How is this related to the maintenance window? Does [downloadonly](https://eclipse.dev/hawkbit/rest-api/targets-api-guide.html#_request_fields_5) mean `deployment.update = skip` and `deployment.maintenanceWindow = available` (or undefined)?

Reply:
Maintenance is communicated with the same DDI parameters (force + skip) . See the corresponding test case for [download only](https://github.com/eclipse/hawkbit/blob/593a0bb146ff49442c78e786ab969b28b599d81a/hawkbit-rest/hawkbit-ddi-resource/src/test/java/org/eclipse/hawkbit/ddi/rest/resource/DdiDeploymentBaseTest.java#L348) and [maintenance](https://github.com/eclipse/hawkbit/blob/593a0bb146ff49442c78e786ab969b28b599d81a/hawkbit-rest/hawkbit-ddi-resource/src/test/java/org/eclipse/hawkbit/ddi/rest/resource/DdiRootControllerTest.java#L654)
`downloadonly` would have maintenance window `undefined` in this case

Reply:
Ah I see.